kubernetes/kubernetes_service.go

```diff
@@ -68,7 +75,11 @@ func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (
 
 	go func() {
 		deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(emptyListOptions)
-		deploymentsChan <- deploymentsResponse{deployments: filterByService(serviceName, deployments), err: err}
+		deployments = filterDeploymentsByService(serviceName, deployments)
+		deploymentsChan <- deploymentsResponse{deployments: deployments, err: err}
+
+		autoscalers, err := in.k8s.AutoscalingV1().HorizontalPodAutoscalers(namespace).List(emptyListOptions)
```

Comment:
Here I see one improvement.
You are querying
```
deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(emptyListOptions)
autoscalers, err := in.k8s.AutoscalingV1().HorizontalPodAutoscalers(namespace).List(emptyListOptions)
```
in the same goroutine, I think you can split them in two, to make the calls concurrently, if not you are waiting and that will introduce latency.

Reply:
Sure! It also have implications in next line :)